src/fsharp/CompileOptions.fs

```diff
@@ -733,6 +733,7 @@ let testFlag tcConfigB =
 // not shown in fsc.exe help, no warning on use, motiviation is for use from VS
 let vsSpecificFlags (tcConfigB: TcConfigBuilder) = 
   [ CompilerOption("vserrors", tagNone, OptionUnit (fun () -> tcConfigB.errorStyle <- ErrorStyle.VSErrors), None, None);
+    CompilerOption("validate-type-providers", tagNone, OptionUnit (id), None, None);  // preserved for compatibility's sake, no longer has any effect
```

Comment:
It's the IDE that passes this flag, it's not something users can control. So I'd prefer not to issue a deprecation warning that the user has no way to address.